core/src/main/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommit.java

```diff
@@ -203,6 +208,7 @@ public void commit() throws CommitException, UnknownTransactionStatusException {
 
       commit.commitRecords(crud.getSnapshot());
       status = Status.COMMITTED;
+      removeActiveTransaction(getId());
```

Comment:
Shouldn't it be removed when CommitException occurs?

Reply:
No. We should call `rollback()` when `CommitException` occurs in `commit()`, and `rollback()` calls this `removeActiveTransaction()`, so it's okay. Thanks.